File: week10/in-class-exercises/python/q1b.py
# Nicholas Ong - Q1(b): Numbers in Files

def solve(output_file_name, n):
    outfile = open(output_file_name, "w")
    outfile.writelines([str(i)+'\n' for i in range(0, n, 2)])
    outfile.close()

def main():
    test_cases = [["q1-output.txt", 10]]
    for filename, n in test_cases:
        solve(filename, n)

if __name__ == "__main__":
    main()

File: week10/in-class-exercises/cpp/q1b.cpp
#include <fstream>
#include <iostream>
#include <string>

void create_number_file(std::string output_file_name, int n)
{
  std::ofstream outfile{output_file_name};

  if(!outfile)
  {
    std::cerr << "Error: could not open file" << std::endl;
    return;
  }

  for (int i=0;i<=n;i=i+2)
  {
    outfile << i << '\n';
  }
  return;
}

int main()
{
  create_number_file("q1-output.txt",10);
}
